active/0000-engines.md

```diff
@@ -0,0 +1,411 @@
+CLI before locking down the public APIs/hooks the router exposes for
+locating and mounting engines. Once this is done, however, we should
+expose and document those hooks so users who cannot use Ember CLI for
+whatever reason can still take advantage of composability.
+
+## Declarative dependencies
+
+The initial scope of declarative dependency sharing is limited in scope to
+services. Should other types of dependencies be declaratively shareable?
+Should addons be the recommended path to share all other dependencies?
+
+## Async mounting of route-less engines
+
+`Route#renderTemplate` is called synchronously, although `Route#mount` should
+surely be async. How async mounting is represented in the route lifecycle is
+TBD. A solution isn't proposed here because the problem is shared by routable
+and async components, and a common solution should be reached.
+
+## Lazy loading manifests
+
+In order to facilitate lazy loading of engines, we will need to determine a
+structure for manifest files that contain an engine's assets. Furthermore, an
+application will need to be configurable with URLs for these manifests.
+
+It's likely that an engine's routing map will always be needed at the time of
+application deployment. Allowing lazy loading of routing maps would prevent the
+formation of any links from a parent application into an engine's routes.
+
+When developed in isolation as addons, engines will have their own sets of
+dependencies. These dependencies will be treated like any other addons when
+engines are deployed together with an application. However, in order to support
+lazy loading, it would be ideal to dedupe dependencies in order to create a lean
+and conflict-free asset manifest.
+
+Reference: deduping strategy discussed by @wycats in
+[this Google doc](https://docs.google.com/a/tomdale.net/document/d/12CsR-zli5oP2TDWOef_-D28zjmbVD83hU4q9_VTk-9s/edit).
+
+## Namespaced access to engine resources
+
+The concept of namespaced access to engine resources is mentioned above as a
+potential goal of a future release of engines. This will require further
+discussion to decide how it should work both technically and semantically, and
+how it applies to lazy-loaded engines.
+
+If these problems can be resolved, this feature would allow for more flexibility
+in parent / engine interactions. Instead of just allowing engines to look up
+resources in a parent, the inverse could also be allowed.
+
+For example, if the `authentication` engine contains
+`engines/authentication/models/user.js`, a parent application could look up this
+same model through a namespace. Perhaps as follows:
+
+```js
+container.lookup('authentication@model:user');
+```
+
+Other APIs in Ember would need to be extended to support namespaces to
+take full advantage of this feature. For example, components that ship
+with an engine might be accessed from the primary application like this:
+
+```handlebars
+{{authentication@login-form obscure-password=true}}
+```
```

Comment:
The original engines RFC included a fair bit about namespacing engines which I shoehorned into this TBD section for this revision. I had the same questions that you raise here. @tomdale and @wycats agreed that an in-depth namespacing discussion needs to happen before this can proceed. This might be best addressed with a separate RFC.